presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestJoinQueries.java

```diff
@@ -2388,6 +2388,18 @@ private void testJoinsWithNullsInternal(Session session)
                 "SELECT * FROM (VALUES 2, 3, null) a(x) " +
                         "FULL OUTER JOIN (VALUES 3, 4, null) b(x) ON a.x = b.x WHERE a.x IS NULL",
                 "SELECT * FROM VALUES (NULL, 4), (NULL, NULL), (NULL, NULL)");
+
+        assertQuerySucceeds(
+                session,
+                "SELECT * FROM customer C INNER JOIN " +
+                        "(SELECT SPLIT(comment, '.') AS k FROM part) P " +
+                        "ON C.name = P.k[2] WHERE CARDINALITY(P.k) >= 2");
```

Comment:
Will be good to also test things like:

ON x[1].y = a[1].b

that's also a dereference.

Reply:
If you want to be really safe, you can check it's just a sequenece of deref eventually ending with a variable so we will only support things like:

ON a.b.c.d = x.y.z

Reply:
@kaikalur 

> x[1].y = a[1].b

I have added new test for this

> a.b.c.d = x.y.z

There is already existing test for this `AbstractTestJoinQueries.testRowFieldAccessorInJoin` . Do you think this would suffice of should this test be enhanced and more cases should be added to this?

Reply:
> @kaikalur
> 
> > x[1].y = a[1].b
> 
> I have added new test for this

I don't see it.

> 
> > a.b.c.d = x.y.z
> 
> There is already existing test for this `AbstractTestJoinQueries.testRowFieldAccessorInJoin` . Do you think this would suffice of should this test be enhanced and more cases should be added to this?

Reply:
The last one is not about testing. It's about making your code even more robust and simply support only simple derefes of the form a.b.c.d and not things like a.b[1].c.d

Reply:
> x[1].y = a[1].b
> I don't see it.

It was probably my mis-interpretation on the test to be added. I earlier added one at L2406. I have now added another test at L2413 that you were asking for.

> The last one is not about testing. It's about making your code even more robust and simply support only simple derefes of the form a.b.c.d and not things like a.b[1].c.d

I understand now. I have now updated it not only to check deref expression at first level. I am going deep and making sure to check there is no subscript / other operator inside.